core/trino-main/src/main/java/io/trino/sql/planner/plan/DynamicFilterSourceNode.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.plan;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.trino.sql.planner.Symbol;
+
+import javax.annotation.concurrent.Immutable;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+@Immutable
+public class DynamicFilterSourceNode
+        extends PlanNode
+{
+    private final PlanNode source;
+    private final Map<DynamicFilterId, Symbol> dynamicFilters;
+    private final boolean isBuildSideReplicated;
```

Comment:
I was using this only to guide thresholds on DF collection in LEP. It occurred to me that we had tuned limits higher for broadcast join because each driver would receive more rows after the exchange. But now that we are collecting before the exchange, this is unnecessary and I got rid of it.